### ISSUE TITLE
local-grpc-client/example LogGrpcInterceptor cannot be injected

### ISSUE THREAD

**Issue by s5364733** [yidongnan/grpc-spring-boot-starter#503]
![image](https://user-images.githubusercontent.com/24503935/110101990-324cb900-7ddf-11eb-9cf2-834bf32b2fbc.png)

The picture shows the original code, when I used it, I used it like this

```java
    @GrpcClient(value = "local-grpc-server", interceptors = LogGrpcInterceptor.class )
    private SimpleBlockingStub simpleStub;

```

### The startup error is as follows：

```
Description:

Field grpcClientService in net.devh.boot.grpc.examples.local.client.GrpcClientController required a bean of type 'net.devh.boot.grpc.examples.local.client.LogGrpcInterceptor' that could not be found.

The injection point has the following annotations:
	- @org.springframework.beans.factory.annotation.Autowired(required=true)


Action:

Consider defining a bean of type 'net.devh.boot.grpc.examples.local.client.LogGrpcInterceptor' in your configuration.


Process finished with exit code 1

```

### When I debugged the following code, I found that there is a problem with the usage of applicationContext as follows：<br/>

```java
  protected List<ClientInterceptor> interceptorsFromAnnotation(final GrpcClient annotation) throws BeansException {
        final List<ClientInterceptor> list = Lists.newArrayList();
        for (final Class<? extends ClientInterceptor> interceptorClass : annotation.interceptors()) {
            final ClientInterceptor clientInterceptor;
            if (this.applicationContext.getBeanNamesForType(ClientInterceptor.class).length > 0) {
                clientInterceptor = this.applicationContext.getBean(interceptorClass);
            } else {
                try {
                    clientInterceptor = interceptorClass.getConstructor().newInstance();
                } catch (final Exception e) {
                    throw new BeanCreationException("Failed to create interceptor instance", e);
                }
            }
            list.add(clientInterceptor);
        }
        for (final String interceptorName : annotation.interceptorNames()) {
            list.add(this.applicationContext.getBean(interceptorName, ClientInterceptor.class));
        }
        return list;
    }

```

1. Among them, this.applicationContext.getBean(interceptorClass)  can only get the specified unique type of BEAN
1. When judging by getBeanNamesForType, the source code uses the interface class, so it is not available here

### We have two solutions can solve the problem:<br/>
1.  Replace  clientInterceptor = this.applicationContext.getBean(interceptorClass) changed to <br/> clientInterceptor = this.applicationContext.getBean(ClientInterceptor) ;

1.
Replaced
```
@Order(Ordered.LOWEST_PRECEDENCE)
@Configuration(proxyBeanMethods = false)
public class GlobalClientInterceptorConfiguration {

    @GrpcGlobalClientInterceptor
    ClientInterceptor logClientInterceptor() {
        return new LogGrpcInterceptor();
    }

}
```

changed to <br/>

```
@Order(Ordered.LOWEST_PRECEDENCE)
@Configuration(proxyBeanMethods = false)
public class GlobalClientInterceptorConfiguration {

    @GrpcGlobalClientInterceptor
    LogGrpcInterceptor logClientInterceptor() {
        return new LogGrpcInterceptor();
    }

}

```

**Comment by ST-DDT** [comment]
Thanks for bringing this to our attention.
I'll check that when I have some spare time.

**Comment by ST-DDT** [comment]
Will be fixed with #532
